azurerm/internal/services/dns/resource_arm_dns_zone.go

```diff
@@ -60,17 +59,6 @@ func resourceArmDnsZone() *schema.Resource {
 				Set:      schema.HashString,
 			},
 
-			"zone_type": {
-				Type:       schema.TypeString,
-				Default:    string(dns.Public),
-				Optional:   true,
-				Deprecated: "Use the `azurerm_private_dns_zone` resource instead.",
-				ValidateFunc: validation.StringInSlice([]string{
-					string(dns.Private),
-					string(dns.Public),
-				}, false),
-			},
```

Comment:
we also need to remove the fields `registration_virtual_network_ids` and `resolution_virtual_network_ids` below / in the data source